src/Mailgun/Api/MailingList.php

```diff
@@ -0,0 +1,268 @@
+use Mailgun\Model\MailingList\UpdateResponse;
+use Mailgun\Model\MailingList\DeleteResponse;
+use Mailgun\Model\MailingList\MembersResponse;
+use Mailgun\Model\MailingList\CreateMemberResponse;
+use Mailgun\Model\MailingList\ShowMemberResponse;
+use Mailgun\Model\MailingList\UpdateMemberResponse;
+use Mailgun\Model\MailingList\DeleteMemberResponse;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * @author Michael Münch <helmchen@sounds-like.me>
+ */
+class MailingList extends HttpApi
+{
+    /**
+     * Returns a paginated list of mailing lists on the domain.
+     *
+     * @param int $limit Maximum number of records to return (optional: 100 by default)
+     *
+     * @return PagesResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function pages($limit = 100)
+    {
+        Assert::integer($limit);
+
+        $params = compact('limit');
+
+        $response = $this->httpGet('/v3/lists/pages', $params);
+
+        return $this->hydrateResponse($response, PagesResponse::class);
+    }
+
+    /**
+     * Creates a new mailing list on the current domain.
+     *
+     * @param string $address      Address for the new mailing list
+     * @param string $name         Name for the new mailing list (optional)
+     * @param string $description  Description for the new mailing list (optional)
+     * @param string $access_level List access level, one of: readonly (default), members, everyone
+     *
+     * @return CreateResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function create($address, $name = null, $description = null, $access_level = 'readonly')
+    {
+        Assert::stringNotEmpty($address);
+        Assert::oneOf($access_level, ['readonly', 'members', 'everyone']);
+
+        $params = compact('address', 'name', 'description', 'access_level');
+
+        $response = $this->httpPost('/v3/lists', $params);
+
+        return $this->hydrateResponse($response, CreateResponse::class);
+    }
+
+    /**
+     * Returns a single mailing list.
+     *
+     * @param string $address Address of the mailing list
+     *
+     * @return ShowResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function show($address)
+    {
+        Assert::stringNotEmpty($address);
+
+        $response = $this->httpGet(sprintf('/v3/lists/%s', $address));
+
+        return $this->hydrateResponse($response, ShowResponse::class);
+    }
+
+    /**
+     * Updates a mailing list.
+     *
+     * @param string $address    Address of the mailing list
+     * @param array  $parameters Array of field => value pairs to update
+     *
+     * @return UpdateResponse|array|ResponseInterface
+     *
+     * @throws \Exception
+     */
+    public function update($address, $parameters = [])
+    {
+        Assert::stringNotEmpty($address);
+        Assert::isArray($parameters);
+
+        foreach ($parameters as $field => $value) {
```

Comment:
Incomplete validation. We know the expected keys for `$parameters` because they are listed on the Mailgun API documentation. We should validate them, to make sure we don't provide wrong key names or wrong values (like subarrays, objects or callables).

Reply:
i extended the switch to include all possible parameters.
unknown `keys` will be catched in the `default` case and simply removed.

Reply:
that sounds good to me - I will think about what exception you take the best, in doubt probably InvalidArgument?

Reply:
Maybe this one, as a generic answer: https://github.com/mailgun/mailgun-php/blob/master/src/Mailgun/Exception/InvalidArgumentException.php

Reply:
perfect! that's what i'am already using in the latest commit, so it should fit.